specs/sharding/beacon-chain.md

```diff
@@ -103,786 +87,558 @@ The following values are (non-configurable) constants used throughout the specif
 ```
 
-#### `get_start_shard`
+#### `low_degree_check`
 
 ```python
-def get_start_shard(state: BeaconState, slot: Slot) -> Shard:
+def low_degree_check(commitments: List[KZGCommitment]):
+    """
+    Checks that the commitments are on a low-degree polynomial
+    If there are 2*N commitments, that means they should lie on a polynomial
+    of degree d = K - N - 1, where K = next_power_of_two(2*N)
+    (The remaining positions are filled with 0, this is to make FFTs usable)
+
+    For details see here: https://notes.ethereum.org/@dankrad/barycentric_low_degree_check
+    """
+    assert len(commitments) % 2 == 0
+    N = len(commitments) // 2
+    r = hash_to_field(commitments)
+    K = next_power_of_two(2 * N)
+    d = K - N - 1
+    r_to_K = pow(r, N, K)
+    roots = roots_of_unity(K)
+
+    # For an efficient implementation, B and Bprime should be precomputed
+    def B(z):
+        r = 1
+        for w in roots[:d + 1]:
+            r = r * (z - w) % MODULUS
+        return r
+
+    def Bprime(z):
+        r = 0
+        for i in range(d + 1):
+            m = 1
+            for w in roots[:i] + roots[i+1:d + 1]:
+                m = m * (z - w) % MODULUS
+            r = (r + M) % MODULUS
+        return r
+
+    coefs = []
+    for i in range(K):
+        coefs.append( - (r_to_K - 1) * modular_inverse(K * roots[i * (K - 1) % K] * (r - roots[i])) % MODULUS)
+    for i in range(d + 1):
+        coefs[i] = (coefs[i] + B(r) * modular_inverse(Bprime(r) * (r - roots[i]))) % MODULUS
+    
+    assert elliptic_curve_lincomb(commitments, coefs) == bls.Z1()
+```
+
+#### `vector_lincomb`
+
+```python
+def vector_lincomb(vectors: List[List[BLSFieldElement]], scalars: List[BLSFieldElement]) -> List[BLSFieldElement]:
     """
-    Return the start shard at ``slot``.
+    Compute a linear combination of field element vectors
     """
-    epoch = compute_epoch_at_slot(Slot(slot))
-    committee_count = get_committee_count_per_slot(state, epoch)
-    active_shard_count = get_active_shard_count(state, epoch)
-    return committee_count * slot % active_shard_count 
+    r = [0 for i in len(vectors[0])]
+    for v, a in zip(vectors, scalars):
+        for i, x in enumerate(v):
+            r[i] = (r[i] + a * x) % MODULUS
+    return [BLSFieldElement(x) for x in r]
 ```
 
-#### `compute_shard_from_committee_index`
+#### `elliptic_curve_lincomb`
 
 ```python
-def compute_shard_from_committee_index(state: BeaconState, slot: Slot, index: CommitteeIndex) -> Shard:
-    active_shards = get_active_shard_count(state, compute_epoch_at_slot(slot))
-    assert index < active_shards
-    return Shard((index + get_start_shard(state, slot)) % active_shards)
+def elliptic_curve_lincomb(points: List[KZGCommitment], scalars: List[BLSFieldElement]) -> KZGCommitment:
+    """
+    BLS multiscalar multiplication. This function can be optimized using Pippenger's algorithm and variants. This is a non-optimized implementation.
+    """
+    r = bls.Z1()
```

Comment:
I assume this is reusing py_ecc idioms.

I'd rather see it spelled `inf_G1()` or `infinity_G1()`